SIPS/sip-29.md

```diff
@@ -0,0 +1,232 @@
+---
+sip: 29
+title: Snap Assets API
+status: Draft
+author: Daniel Rocha (@danroc), Guillaume Roux (@GuillaumeRx)
+created: 2024-12-05
+---
+
+## Abstract
+
+This SIP aims to define a new API that can be exposed by Snaps to allow clients
+to retrieve asset information in a chain-agnostic way.
+
+## Motivation
+
+To enable clients to be chain-agnostic, the logic for obtaining asset
+information should be abstracted away from the client. Additionally, this SIP
+defines the types that represent the asset information required by clients.
+
+## Specification
+
+> Indented sections like this are considered non-normative.
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written
+in uppercase in this document are to be interpreted as described in [RFC
+2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Definitions
+
+1. In this document, all definitions are written in TypeScript.
+
+2. Any time an asset needs to be identified, it MUST use the [CAIP-19][caip-19]
+   representation.
+
+### Snap Manifest
+
+This SIP introduces a new permission named `endowment:assets`.
+This permission grants a Snap the ability to provide asset information to the client.
+
+This permission is specified as follows in `snap.manifest.json` files:
+
+```json
+{
+  "initialPermissions": {
+    "endowment:assets": {
+      "scopes": [
+        "bip122:000000000019d6689c085ae165831e93"
+      ]
+    }
+  }
+}
+```
+
+`scopes` - A non-empty array of CAIP-2 chain IDs that the Snap supports. This field is useful for a client in order to avoid unnecessary overhead.
```

Comment:
> to avoid unnecessary overhead.

Since this information would be static, will the Snap only be "restricted" to those scopes? Or can Snap also returns assets informations for other non-declared scopes?

Reply:
My expectation for this would be that on the controller end we would only query this Snap for assets if we are interested in the scopes that it has.

Reply:
Indeed we should enforce that the Snap can return info only for assets under its scope. (This is an implementation detail of the controller.)